experiments/AMIP/modular/components/atmosphere/climaatmos_init.jl

```diff
@@ -1,98 +1,133 @@
+    parent(integrator.p.ᶠradiation_flux) .= parent(zeros(axes(integrator.p.ᶠradiation_flux)))
+
+    ClimaAtmosSimulation(integrator.p.params, Y, spaces, integrator)
 end
 
 # required by the Interfacer
-get_field(sim::ClimaAtmosSimulation, ::Val{:radiative_energy_flux}) = level(sim.integrator.p.ᶠradiation_flux, half)
+get_field(sim::ClimaAtmosSimulation, ::Val{:radiative_energy_flux}) =
+    Fields.level(sim.integrator.p.ᶠradiation_flux, half)
 get_field(sim::ClimaAtmosSimulation, ::Val{:liquid_precipitation}) =
     sim.integrator.p.col_integrated_rain .+ sim.integrator.p.col_integrated_snow # all fallen snow melts for now
 get_field(sim::ClimaAtmosSimulation, ::Val{:snow_precipitation}) = sim.integrator.p.col_integrated_snow .* FT(0)
 
-get_field(sim::ClimaAtmosSimulation, ::Val{:turbulent_energy_flux}) = sim.integrator.p.ρ_dif_flux_h_tot
-get_field(sim::ClimaAtmosSimulation, ::Val{:turbulent_moisture_flux}) = sim.integrator.p.ρ_dif_flux_q_tot
+get_field(sim::ClimaAtmosSimulation, ::Val{:turbulent_energy_flux}) =
+    Geometry.WVector.(sim.integrator.p.sfc_conditions.ρ_flux_h_tot)
+get_field(sim::ClimaAtmosSimulation, ::Val{:turbulent_moisture_flux}) =
+    Geometry.WVector.(sim.integrator.p.sfc_conditions.ρ_flux_q_tot)
 
 function update_field!(sim::ClimaAtmosSimulation, ::Val{:surface_temperature}, field)
     sim.integrator.p.radiation_model.surface_temperature .= RRTMGPI.field2array(field)
-    parent(sim.integrator.p.T_sfc) .= parent(field)
-
 end
 function update_field!(sim::ClimaAtmosSimulation, ::Val{:albedo}, field)
     sim.integrator.p.radiation_model.diffuse_sw_surface_albedo .=
         reshape(RRTMGPI.field2array(field), 1, length(parent(field)))
     sim.integrator.p.radiation_model.direct_sw_surface_albedo .=
         reshape(RRTMGPI.field2array(field), 1, length(parent(field)))
 end
-function update_field!(sim::ClimaAtmosSimulation, ::Val{:roughness_momentum}, field)
-    parent(sim.integrator.p.sfc_inputs.z0m) .= parent(field)
-end
-function update_field!(sim::ClimaAtmosSimulation, ::Val{:roughness_buoyancy}, field)
-    parent(sim.integrator.p.sfc_inputs.z0b) .= parent(field)
-end
 
 step!(sim::ClimaAtmosSimulation, t) = step!(sim.integrator, t - sim.integrator.t, true)
 
 reinit!(sim::ClimaAtmosSimulation) = reinit!(sim.integrator)
 
+"""
+    update_sim!(atmos_sim::ClimaAtmosSimulation, csf)
+
+Updates the surface fields for temperature and albedo.
+
+# Arguments
+- `atmos_sim`: [ClimaAtmosSimulation] containing an atmospheric model simulation object.
+- `csf`: [NamedTuple] containing coupler fields.
+"""
+function update_sim!(atmos_sim::ClimaAtmosSimulation, csf, turbulent_fluxes)
+    update_field!(atmos_sim, Val(:albedo), csf.albedo)
+    update_field!(atmos_sim, Val(:surface_temperature), csf.T_S)
+end
+
 
 # flux calculation borrowed from atmos
+"""
+    CoupledMoninObukhov()
+A modified version of a Monin-Obukhov surface for the Coupler, see the link below for more information
```

Comment:
Why does this need to be modified for the coupler?

Reply:
It's an extension that's unique to the coupler setup rn (i.e. passing in fields rather than floats/functions for the surface states), so it's best suited here.